rover_gui/src/Global/Helpers/QToastNotification.cpp

```diff
@@ -0,0 +1,431 @@
+                icon = QApplication::style()->standardIcon(QStyle::SP_MessageBoxCritical);
+                break;
+
+            case QToastNotification::eNotifType::SUCCESS:
+                icon = QApplication::style()->standardIcon(QStyle::SP_DialogApplyButton);
+                break;
+
+            default:
+                icon = QApplication::style()->standardIcon(QStyle::SP_MessageBoxQuestion);
+                break;
+        }
+
+        _ui.iconSlot->setIcon(icon);
+        _ui.iconSlot->setIconSize(QSize(40, 40));
+        _ui.textErrorMessage->setText(description_);
+        _ui.titleLineEdit->setText(title_);
+
+        this->adjustSize();
+
+        size_t X = targetScreenRect.right() - width() - MARGIN_NOTIF;
+        size_t startY = targetScreenRect.bottom() - height() + 2 * MARGIN_NOTIF;
+        size_t endY = targetScreenRect.bottom() - height() - 2 * MARGIN_NOTIF;
+
+        _slideInAnim.setStartValue(QPoint(X, startY));
+        _slideInAnim.setEndValue(QPoint(X, endY));
+
+        _slideOutAnim.setStartValue(QPoint(X, endY));
+        _slideOutAnim.setEndValue(QPoint(X, startY));
+
+        move(X, endY);
+        setWindowOpacity(0.0);
+        _ui.progressBar->setValue(100);
+
+        this->show();
+        this->raise();
+
+        _progressBarAnim.setDuration(durationMs_);
+
+        _fadeInAnim.start();
+        _slideInAnim.start();
+        _progressBarAnim.start();
+
+        _closeTimer.start(durationMs_);
+
+        QTime currentTime = QTime::currentTime();
+
+        sNotificationInfo data = {currentTime, title_, description_, type_};
+        this->saveNotifInfo(data);
+    }
+
+    void QToastNotification::hideNotification()
+    {
+        _fadeInAnim.stop();
+        _slideInAnim.stop();
+        _progressBarAnim.stop();
+        _closeTimer.stop();
+
+        _fadeOutAnim.start();
+        _slideOutAnim.start();
+    }
+
+    void QToastNotification::saveNotifInfo(const sNotificationInfo& info_)
+    {
+        history.push_back(info_);
+
+        if (history.size() >= HISTORY_MAX_SIZE)
+        {
+            history.pop_front();
+        }
+    }
+
+    QNotificationShowHistory::QNotificationShowHistory()
+    {
+        setWindowFlags(Qt::FramelessWindowHint | Qt::ToolTip);
+        setAttribute(Qt::WA_TranslucentBackground);
+
+        _ui.setupUi(this);
+
+        _scrollAreaContainer.setLayout(&_scrollAreaLayout);
+        _scrollAreaContainer.setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
+        _scrollAreaContainer.setMinimumWidth(_ui.historyScrollArea->width());
+
+        _scrollAreaLayout.setAlignment(Qt::AlignTop);
+
+        this->move(-100, height());
+
+        _ui.historyScrollArea->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+        _ui.historyScrollArea->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+        _ui.historyScrollArea->setVisible(false);
+
+        _targetScreenRect = QToastNotification::getInstance().targetScreenRect;
```

Comment:
should use a get method:
```suggestion
        _targetScreenRect = QToastNotification::getInstance().getTargetScreenRect();
```

Reply:
done